tests/stdlib/tasynchttpserver_transferencoding.nim

```diff
@@ -0,0 +1,155 @@
+        echo "simpleEncodingTest ok"
+        await request.respond(Http200, "Good")
+
+    let server = newAsyncHttpServer()
+    discard server.serve(Port(64123), handler)
+
+    let data = postBegin & exampleWorld
+
+    var socket = newSocket()
+    socket.connect("127.0.0.1", Port(64123))
+    socket.send(data)
+
+    waitFor sleepAsync(10)
+
+    socket.close()
+    server.close()
+
+proc testSimpleChunks() =
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding
+    let exampleMozilla = ("7\r\n" &
+                         "Mozilla\r\n" &
+                         "9\r\n" &
+                         "Developer\r\n" &
+                         "7\r\n" &
+                         "Network\r\n" &
+                         "0\r\n" &
+                         "\r\n")
+    let expected = "MozillaDeveloperNetwork"
+
+    proc handler(request: Request) {.async.} =
+        doAssert(request.body == expected)
+        doAssert(request.headers.hasKey("Transfer-Encoding"))
+        doAssert(not request.headers.hasKey("Content-Length"))
+        sanitySimpleChunksTest = true
+        echo "simpleChunksTest ok"
+        await request.respond(Http200, "Good")
+
+    let server = newAsyncHttpServer()
+    discard server.serve(Port(64123), handler)
+
+    let data = postBegin & exampleMozilla
+
+    var socket = newSocket()
+    socket.connect("127.0.0.1", Port(64123))
+    socket.send(data)
+
+    waitFor sleepAsync(10)
+
+    socket.close()
+    server.close()
+
+proc testComplexChunks() =
+    # https://en.wikipedia.org/wiki/Chunked_transfer_encoding#Example
+    let exampleWikipedia = ("4\r\n" &
+                            "Wiki\r\n" &
+                            "6\r\n" &
+                            "pedia \r\n" &
+                            "E\r\n" &
+                            "in \r\n" &
+                            "\r\n" &
+                            "chunks.\r\n" &
+                            "0\r\n" &
+                            "\r\n")
+    let expected = "Wikipedia in \r\n\r\nchunks."
+
+    proc handler(request: Request) {.async.} =
+        doAssert(request.body == expected)
+        doAssert(request.headers.hasKey("Transfer-Encoding"))
+        doAssert(not request.headers.hasKey("Content-Length"))
+        sanityComplexChunksTest = true
+        echo "complexChunksTest ok"
+        await request.respond(Http200, "Good")
+
+    let server = newAsyncHttpServer()
+    discard server.serve(Port(64123), handler)
+
+    let data = postBegin & exampleWikipedia
+
+    var socket = newSocket()
+    socket.connect("127.0.0.1", Port(64123))
+    socket.send(data)
+
+    waitFor sleepAsync(10)
+
+    socket.close()
+    server.close()
+
+testSimpleHelloWorld()
+testSimpleEncoding()
+testSimpleChunks()
+testComplexChunks()
+
+doAssert(sanitySimpleHelloWorldTest)
+doAssert(sanitySimpleEncodingTest)
+doAssert(sanitySimpleChunksTest)
+doAssert(sanityComplexChunksTest)
+echo "Ok"
```

Comment:
remove this line

Reply:
Why remove it? I think it should be re-added in fact. It's useful to get a message that the test went well, otherwise you might mistake a success for a missed `-r` flag to `nim` :)

Reply:
At first, I was in favour of keeping the echo, but I had issues figuring out how to get it to work with the test suite (issues about the `megatest` failing). However, I feel comfortable removing it because each test also sets a bool to true, and at the end we assert that this bool is true. I think this fills the exact same use case as the echos, and is also programmatically verified by the `doAssert(sanity...Test)` at the end.

Reply:
things like `echo "Ok"` are pointless and should be removed (and there's no reason to single this test out, most other tests (at least new ones) don't have such trailing echo). Tests are run (not just compiled) by default so there's no need to check that the test actually ran. If some future regression in test suite makes it so that no test is actually run, we'd know soon enough (and in fact I'm pretty sure it'd fail the test suite).

> I had issues figuring out how to get it to work with the test suite (issues about the megatest failing)

that's because with echo "OK", you'd also need to add a testament spec section with output you expect as megatest gathers all expected/actual outputs. But again, I agree with @xflywind, we shouldn't use `echo "OK"` here.